packages/flutter/test/material/back_button_test.dart

```diff
@@ -180,4 +180,35 @@ void main() {
     ));
     expect(iconText.text.style.color, Colors.red);
   });
+
+  testWidgets('CloseButton onPressed overrides default pop behavior', (WidgetTester tester) async {
+    bool closePressed = false;
+    await tester.pumpWidget(
+      MaterialApp(
+        home: const Material(child: Text('Home')),
+        routes: <String, WidgetBuilder>{
+          '/next': (BuildContext context) {
+            return Material(
+              child: Center(
+                child: CloseButton(onPressed: () => closePressed = true),
+              ),
+            );
+          },
+        },
+      ),
+    );
+
+    tester.state<NavigatorState>(find.byType(Navigator)).pushNamed('/next');
+
+    await tester.pumpAndSettle();
+
```

Comment:
Just some extra checks before calling tap to ensure the state prior to the tap is consistent with expectations
```suggestion
    expect(find.text('Home'), findsNothing); // Start off on the second page.
    expect(closePressed, false); // closePressed should still be false.
```